File: solutions/problem_1/solution.go
package main

import (
  "fmt"
  "time"
)

func divisibleBy(num, divisor int) bool {
  return num % divisor == 0
}

func SumMultipleOfThreeAndFive() (sum int) {
  sum = 0
  for i := 0; i < 1000; i++ {
    if divisibleBy(i, 3) || divisibleBy(i, 5) {
      sum += i
    }
  }
  return
}

func main() {
  start := time.Now()
  result := SumMultipleOfThreeAndFive()
  elapsed := time.Since(start)
  fmt.Printf("=> Result: %d\n", result)
  fmt.Printf("=> Time: %.8fs\n", elapsed.Seconds())
}

File: solutions/problem_1/solution.pl
use Time::HiRes qw( time );

sub divisible_by {
  my ($y, $z) = @_;
  $y % $z == 0
}

sub solution {
  my ($n1, $n2, $limit) = @_;
  my @numbers;

  for $n (1..$limit-1) {
    if(divisible_by($n, $n1) || divisible_by($n, $n2)){
      push @numbers, $n;
    }
  }

  my $total = 0;
  $total += $_ for @numbers;

  $total;
}

my $start = time();
my $output = solution(3, 5, 1000);
my $end = time();

print "=> Result: ", $output, "\n";
printf "=> Time: %fs\n", ($end - $start);